pandas/core/categorical.py

```diff
@@ -336,11 +336,26 @@ def copy(self):
                            categories=self.categories, ordered=self.ordered,
                            fastpath=True)
 
-    def astype(self, dtype):
-        """ coerce this type to another dtype """
+    def astype(self, dtype, copy=True):
+        """
+        Coerce this type to another dtype
+
+        Parameters
+        ----------
+        dtype : numpy dtype or pandas type
+        copy : bool, default True
+            If copy is set to False and dtype is categorical, the original
```

Comment:
yeah, this is right, though trying to think of a more user-friendly wording. Essentially we copy if ``True``, otherwise we *still* may copy, but if dtype compat then we wont'.
From numpy astype (maybe for some inspiration).
```
copy : bool, optional
    By default, astype always returns a newly allocated array. If this
    is set to false, and the `dtype`, `order`, and `subok`
    requirements are satisfied, the input array is returned instead
    of a copy.
```